CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc

```diff
@@ -479,7 +480,25 @@ void SiStripGainFromCalibTree::bookDQMHistos(const char* dqm_dir, const char* ta
 
     int elepos = (m_harvestingMode && AlgoMode=="PCL")? Harvest : statCollectionFromMode(tag);
 
-    Charge_Vs_Index[elepos]           = dbe->book2S(cvi.c_str()     , cvi.c_str()     , 88625, 0   , 88624,2000,0,4000);
+    //setting up the bin array for the Charge_Vs_Index histogram
+    float* binXarray = new float[NStripAPVs+1];
+    for(int a=0;a<=NStripAPVs;a++){
+       binXarray[a] = (float)a;
+    }
+ 
+    float* binYarray = new float[688];
```

Comment:
hi @dimattia - please use std:array<float,.. > both here and above. [perhaps binXarray has to be a vector if NStripAPVs is not a const expr]